src/pepr/policies/security.ts

```diff
@@ -16,6 +16,62 @@ import {
 import { exemptionAnnotationPrefix, isExempt, markExemption } from "./exemptions";
 
 const { containers } = sdk;
+
+/**
+ * This policy restricts the use of the Istio proxy user/group (1337) to only be used by Istio proxy containers.
+ * It allows specific Istio components (waypoints, ztunnel, and sidecars) to use these IDs.
+ * This prevents unauthorized pods from running with elevated privileges that could be used to bypass security controls.
+ */
+When(a.Pod)
+  .IsCreatedOrUpdated()
+  .Mutate(markExemption(Policy.RestrictIstioUser))
+  .Validate(request => {
+    if (isExempt(request, Policy.RestrictIstioUser)) {
+      return request.Approve();
+    }
+
+    const pod = request.Raw as V1Pod;
+    const podSpec = pod.spec || ({} as V1PodSpec);
+    const podSecurityCtx = podSpec.securityContext || ({} as V1PodSecurityContext);
+
+    // Check pod-level security context for UID/GID 1337
+    if (
+      podSecurityCtx.runAsUser === 1337 ||
+      podSecurityCtx.runAsGroup === 1337 ||
+      podSecurityCtx.fsGroup === 1337 ||
+      podSecurityCtx.supplementalGroups?.includes(1337)
+    ) {
+      return request.Deny(
+        "Pods cannot use UID/GID 1337 (Istio proxy) unless they are trusted Istio components",
+      );
+    }
+
+    // Check container security contexts
+    for (const container of containers(request)) {
+      const containerCtx = container.securityContext || {};
+
+      // Check if this is an Istio proxy container
+      const isIstioProxy =
+        container.name === "istio-proxy" &&
+        container.ports?.some(p => p.name === "http-envoy-prom") &&
+        container.args?.some(arg => arg.includes("proxy"));
+
+      // Only check UID/GID 1337 if this is not an Istio proxy container
+      if (!isIstioProxy) {
+        const runAsUser = containerCtx.runAsUser ?? podSecurityCtx.runAsUser;
+        const runAsGroup = containerCtx.runAsGroup ?? podSecurityCtx.runAsGroup;
+
+        if (runAsUser === 1337 || runAsGroup === 1337) {
```

Comment:
Another nit but the nullish check here isn't really necessary since we check the pod securityContext above.

```suggestion
        if (containerCtx.runAsUser === 1337 || containerCtx.runAsGroup === 1337) {
```